Fixes for Cloud-Native eBook issues (#17827)
dotnet/docs: docs/architecture/cloud-native/combine-containers-serverless-approaches.md

## When does it make sense to use containers with serverless?

Your Azure Function has no knowledge of the platform on which it's deployed. For some scenarios, you may have specific requirements and need to customize the environment on which your function code will run. You'll need a custom image that supports dependencies or a configuration not supported by the default image. In these cases, it makes sense to deploy your function in a custom Docker container.

## When should you avoid using containers with Azure Functions?

If you want to use consumption billing, you won't be able to run your function in a container. What's more, if you deploy your function to a Kubernetes cluster, you'll no longer benefit from the built-in scaling provided by Azure Functions. You'll need to use Kubernetes' scaling features, described earlier in this chapter.

## How to combine serverless and Docker containers

To wrap an Azure Function in a Docker container, install the [Azure Functions Core Tools](https://github.com/Azure/azure-functions-core-tools) and then run the following command:

```console
func init ProjectName --worker-runtime dotnet --docker
```

> Once created, the function will include a Dockerfile. You can create and test your function locally. When deploying a function in containers, the ritual of building an image and hosting a container isn't necessary. You write your function code, include a Docker file, and trigger the function. It deploys without additional overhead or configuration.

Reviewer: ```suggestion
When the project is created, it will include a Dockerfile and the worker runtime configured to `dotnet`. Now, you can create and test your function locally. Build and run it using the  `docker build` and `docker run` commands. For detailed steps to get started building Azure Functions with Docker support, see the [Create a function on Linux using a custom image](https://docs.microsoft.com/azure/azure-functions/functions-create-function-linux-custom-image) tutorial.
```

> When creating Azure functions, you'll want to reference version 3 of the Azure Functions runtime, which targets .NET Core 3.0 and later. [This link](https://docs.microsoft.com/azure/azure-functions/functions-versions) describes the runtime versions available for Azure Functions.

Reviewer: ```suggestion
```

> For detailed steps to get started building Azure Functions with Docker support, see the [Create a function on Linux using a custom image](https://docs.microsoft.com/azure/azure-functions/functions-create-function-linux-custom-image) tutorial.

Reviewer: ```suggestion
```

Choose which worker runtime you want from the following options:

> - `Dotet` (C# or F#)

Reviewer: typo use `dotnet`

Reviewer: Since, we're focusing on dotnet through out this book, I suggest to use the dotnet runtime. I couldn't modify the unedited text here, so I created a gist for your reference: https://gist.github.com/nishanil/27687b0a6255d46c87c66c528c0e0f23

Author: Fixed. Guess each platform should be capitalized? 'Dotnet,' not 'dotnet'??

> - `node` (JavaScript)

Reviewer: Per https://github.com/dotnet/docs/issues/17529, you may want to mention the presence of more runtime availability? IMO, instead of mentioning all the runtimes, it may be worth adding "and more" as text and reference it to the link mentioned in the issue.

Author: Added
